src/config/configuration.f90

```diff
@@ -300,6 +305,33 @@ subroutine copy_array_i(input, output, invalid)
         output(:n)  = input(:n)
     end subroutine copy_array_i
 
+    subroutine copy_array_weights(input, output, n, default)
+        implicit none
+        real, dimension(:), intent(in)   :: input
+        real, dimension(:), allocatable, intent(inout):: output
+        integer, intent(in)   :: n
+        real, optional :: default
+        integer :: i
+        real    :: default_test
+
+        if (present(default)) then
```

Comment:
Unless you had something in mind with this, I would probably remove the default value. The default is already set outside of this, it should just be copying a given array while handling resizing issues.